executor/converter/convert_stmt.go

```diff
@@ -112,9 +112,11 @@ func convertDelete(converter *expressionConverter, v *ast.DeleteStmt) (*stmts.De
 		return nil, errors.Trace(err)
 	}
 	oldDelete.Refs = oldRefs
-	for _, val := range v.Tables {
-		tableIdent := table.Ident{Schema: val.Schema, Name: val.Name}
-		oldDelete.TableIdents = append(oldDelete.TableIdents, tableIdent)
+	if v.Tables != nil {
```

Comment:
For single-table mode, it is nil. See: https://github.com/pingcap/tidb/blob/master/parser/parser.y#L1199